bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/LinkyAuthServlet.java

```diff
@@ -0,0 +1,176 @@
+     * codes via the url and these are processed. In case of an error this is shown to the user. If the user was
+     * authorized this is passed on to the handler. Based on all these different outcomes the HTML is generated to
+     * inform the user.
+     *
+     * @param replaceMap a map with key String values that will be mapped in the HTML templates.
+     * @param servletBaseURL the servlet base, which should be used as the Linky redirect_uri value
+     * @param queryString the query part of the GET request this servlet is processing
+     */
+    private void handleLinkyRedirect(Map<String, String> replaceMap, String servletBaseURL,
+            @Nullable String queryString) throws LinkyException {
+        replaceMap.put(KEY_AUTHORIZED_USER, "");
+        replaceMap.put(KEY_ERROR, "");
+        replaceMap.put(KEY_PAGE_REFRESH, "");
+
+        if (queryString != null) {
+            final MultiMap<String> params = new MultiMap<>();
+            UrlEncoded.decodeTo(queryString, params, StandardCharsets.UTF_8.name());
+            final String reqCode = params.getString("code");
+            final String reqState = params.getString("state");
+            final String reqError = params.getString("error");
+
+            replaceMap.put(KEY_PAGE_REFRESH, "");
+
+            // params.isEmpty() ? "" : String.format(HTML_META_REFRESH_CONTENT, servletBaseURL)
+
+            if (!StringUtil.isBlank(reqError)) {
+                logger.debug("Linky redirected with an error: {}", reqError);
+                replaceMap.put(KEY_ERROR, String.format(HTML_ERROR, reqError));
+            } else if (!StringUtil.isBlank(reqState)) {
+                try {
+                    replaceMap.put(KEY_AUTHORIZED_USER, String.format(HTML_USER_AUTHORIZED,
+                            reqCode + " / " + apiBridgeHandler.authorize(servletBaseURL, reqState, reqCode)));
+                } catch (RuntimeException e) {
+                    logger.debug("Exception during authorizaton: ", e);
+                    replaceMap.put(KEY_ERROR, String.format(HTML_ERROR, e.getMessage()));
+                }
+            }
+        }
+    }
+
+    /**
+     * Replaces all keys from the map found in the template with values from the map. If the key is not found the key
+     * will be kept in the template.
+     *
+     * @param template template to replace keys with values
+     * @param map map with key value pairs to replace in the template
+     * @return a template with keys replaced
+     */
+    private String replaceKeysFromMap(String template, Map<String, String> map) {
+        final Matcher m = MESSAGE_KEY_PATTERN.matcher(template);
+        final StringBuffer sb = new StringBuffer();
+
+        while (m.find()) {
+            try {
+                final String key = m.group(1);
+                m.appendReplacement(sb, Matcher.quoteReplacement(map.getOrDefault(key, "${" + key + '}')));
+            } catch (RuntimeException e) {
```

Comment:
What raises a `RuntimeException` ?

Reply:
no one, I've replaced it by LinkyException are there is already a try/catch clause inside apiBridgeHandler.authorize()